docs/agents/agent-development.md

```diff
@@ -448,6 +448,12 @@ There is no regex currently suggested for cgroup-v2. Look in other agent PRs to
 
 Whenever agents fail to discover the memory mount path, they should default to `/sys/fs/cgroup/memory`.
 
+#### Special values for unlimited memory quota
+
+There is a special value used to indicate that the cgroup is not configured with a memory limit. In cgroup v1, this value is numeric - `0x7ffffffffffff000` and in cgroup v2 it is represented by the string `max`. 
+Agents should not send the `system.process.cgroup.memory.mem.limit.bytes` metric when the read value is `max`. 
+They may or may not send this metric when reading `0x7ffffffffffff000`, as the UI is handling this case.
```

Comment:
If agents already handle the special value `max`, why not also handle `0x7ffffffffffff000`?

> as the UI is handling this case

This is logic specific to the APM UI. Users trying to create their own visualizations will still face that issue.

Reply:
> If agents already handle the special value max, why not also handle 0x7ffffffffffff000?

As written in the spec - agents may or may not send it. The UI will add this simple check because it's very easy and already agreed upon and can do no harm if they do.

> Users trying to create their own visualizations will still face that issue.

They will have face this issue either way - either handle cases where the limit metric does not exist or handle cases of special unlimited value. 

Do you prefer we force all agents to never send this metric when they read any special unlimited value?